### DISCUSSION TITLE
Fee calculation based on payloadType

### DISCUSSION THREAD

**Discussion by ZigBalthazar** [pactus-project/pactus#612] [General]
## Description

This article discusses how to determine fees for different types of payloads. For instance, when calculating fees for transactions with an unbound payload type, the fee is considered to be zero. This raises the question of whether such transactions should have a zero fee. Additionally, we explore how transaction fees for smart contracts will be calculated. Could an Ethereum-like fee calculation model be appropriate for this situation?

**Comment by b00f** [comment]
This is an important discussion and will help us to decide and finalize the fee model in Pactus.
To understand more about the fee model in Pactus please read here: https://pactus.org/learn/transaction/fee/

**Comment by kehiy** [comment]
@b00f @ZigBalthazar we are close to the main net and we can't change the fee calculation then easily. so, should we continue discussion or close this?

**Reply by b00f** [reply]
This discussion is related to the this issue: https://github.com/pactus-project/pactus/issues/493

Imagine a user who intends to withdraw their entire balance. Let's do some math.

Let's denote 
- $T$ as total amount, 
- $F_{min}$ as maximum fee 
- $F_{max}$ as maximum fee 
- $f$ as percentage fee and $P$ as percentage in fee calculation that explained [fee parameters](https://pactus.org/learn/transaction/fee/) . 

We want to calculate the value of $a$ such that $T = a + f$.

Starting with:

$$T = a + f$$

And knowing that $f = (P \times a)$, we get:

$$T = a + (P \times a)$$

By factoring out $a$:

$$T = a \times (P + 1)$$

Therefore:

$$a = \frac{T}{(P + 1)}$$

For instance, if a user wishes to withdraw 1000 coins with a fee percentage $P$ set at 0.0001, the maximum withdrawal amount would be:

$a = \frac{1000}{(0.0001 + 1)} = 999.900,009,999$ (rounded to 9 decimal places).

In this calculation, we haven't considered the minimum and maximum fee parameters. I invite you to think about it, @ZigBalthazar and @kehiy.

**Reply by kehiy** [reply]
@b00f So, what's the problem with this model of fee calculation? I think this is a good approach for the calculation of fees. if we have a problem with min and max amounts, we can always check if we reached them, just get back to the min or max.

**Reply by b00f** [reply]
There is no problem. We need to provide an API for user to calculate maximum amount for withdrawal.

**Reply by kehiy** [reply]
@b00f Based on the title and first message of tread, I thought it was about the fee calculation model(?).
about calculating max amount for withdrawal, that doesn't seem to be a hard thing to do. we can add the function GUI button and command line feature by fixing the mentioned issue.

**Reply by kehiy** [reply]
have a look at first message.

**Comment by kehiy** [comment]
@b00f @ZigBalthazar 
We decided to close this, there is no need to change the fee calculation model. Also about the gas fee, we will discuss it later in detail, and no problems with this model for then. also about ALL buttons in wallets (GUI/CLI), we close it by fixing #493.
